### PR TITLE
Sum of vectors

### PR DESCRIPTION
**Pull request by Olgerd96**

### CODE REVIEW

File: 01f_work_with_vector_int.cpp [hunk 1]
```diff
@@ -33,15 +33,36 @@ void print_stats(const std::vector<int> input_object)
...
+            input1[i] += i;
+        }
+        else
+        {
+            input2[i] -= i;
+        }
+    }
     print_vector(input1);
+    print_vector(input2);
+    print_vector(vectors_sum(input1, input2));
```
Comment (shssf) [inline 01f_work_with_vector_int.cpp]: Please do not use several operations at single line. It has many disadvantages.

File: 01f_work_with_vector_int.cpp [hunk 2]
```diff
@@ -33,15 +33,36 @@ void print_stats(const std::vector<int> input_object)
...
+    for (int i=0; i < std::max(input1.size(), input2.size()); i++){
+        if (i%2 == 0)
+        {
+            input1[i] += i;
+        }
+        else
+        {
+            input2[i] -= i;
+        }
+    }
```
Comment (shssf) [inline 01f_work_with_vector_int.cpp]: no need this block for this time

**Review by shssf — APPROVED** [review]

File: 01f_work_with_vector_int.cpp [hunk 3]
```diff
@@ -1,47 +1,65 @@
...
-    input1.push_back(99);
-    print_vector(input1);
+  return sum_vector;
+}
+
+int main() {
+  const int vector_size = 268435456;
+  std::vector<int> input1(vector_size, 42);
+  std::vector<int> input2(vector_size, 53);
+  std::vector<int> input_sum;
```
Comment (shssf) [inline 01f_work_with_vector_int.cpp]: it is better to name it `result`

File: 01f_work_with_vector_int.cpp [hunk 4]
```diff
@@ -1,47 +1,65 @@
...
+  return sum_vector;
+}
+
+int main() {
+  const int vector_size = 268435456;
+  std::vector<int> input1(vector_size, 42);
+  std::vector<int> input2(vector_size, 54);
+  std::vector<int> input_sum;
+
+  for (size_t i = 0; i < std::max(input1.size(), input2.size()) && i < 20; i++) {
```
Comment (shssf) [inline 01f_work_with_vector_int.cpp]: `) && i < 20;`
it is a bit unclear and strongly depends on operators priority.
it is always better to explicitly put priority by brackets like
`) && (i < 20);`

Comment (Olgerd96) [reply 01f_work_with_vector_int.cpp]: fixed by https://github.com/Olgerd96/cpp/commit/2de398898e821caa12457a67d9d91c47074afc91

File: 01f_work_with_vector_int.cpp (hunk 3 above)
Comment (Olgerd96) [reply 01f_work_with_vector_int.cpp]: fixed by https://github.com/Olgerd96/cpp/commit/d64aa2d1f9f11396ac84ec0dc6000d8a386b3f9b

File: 01f_work_with_vector_int.cpp (hunk 2 above)
Comment (Olgerd96) [reply 01f_work_with_vector_int.cpp]: to sad look at same numbers... i modife just few at begining - just for fun. 
https://github.com/Olgerd96/cpp/commit/8ad2502d76be87356763cdb7a01e3db19347188d

File: 01f_work_with_vector_int.cpp (hunk 1 above)
Comment (Olgerd96) [reply 01f_work_with_vector_int.cpp]: fixed by https://github.com/Olgerd96/cpp/commit/d64aa2d1f9f11396ac84ec0dc6000d8a386b3f9b

**Pull request merged**